src/model/MqttAction.ts

```diff
@@ -143,33 +152,31 @@ class MqttAction implements Action {
         );
 
         client.on('connect', () => {
-            getLogger(scenario.name).debug(
+            logDebug(
                 `MQTT connection to ${this.url} successfully opened for ${this.durationInSec}s`,
-                ctx,
             );
-            client.subscribe(this.topic, (error: any, granted: any) => {
+            client.subscribe(this.topic, (error, granted) => {
                 if (error) {
-                    getLogger(scenario.name).error(
+                    logError(
                         `Error while subscribing to ${this.topic}: ${error}`,
-                        ctx,
                     );
+                    reject();
                 } else {
-                    getLogger(scenario.name).debug(
+                    logDebug(
                         `Successfully subscribed to '${granted[0].topic}' (qos: ${granted[0].qos})`,
-                        ctx,
                     );
                 }
             });
 
             setTimeout(() => client.end(), this.durationInSec * 1000);
         });
 
-        client.on('message', (topic: any, message: any) => {
+        client.on('message', (topic, message) => {
             let msgObj = {};
 
             if (messageType === 'json') {
                 msgObj = JSON.parse(message.toString());
-            } else if (messageType == 'proto') {
+            } else if (messageType === 'proto') {
```

Comment:
(optional): add MessageType enum

Reply:
Doing this would also require us to validate user input (i.e. the used type in the action definition) before constructing the action. This is the correct thing to do, but out of scope here. I'll create an issue for that.